extensions/notebook/src/book/bookTreeView.ts

```diff
@@ -223,10 +223,10 @@ export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeIte
 				await this.bookTocManager.recovery();
 				vscode.window.showErrorMessage(loc.editBookError(updateBook.book.contentPath, e instanceof Error ? e.message : e));
 			} finally {
-				this.fireBookRefresh(targetBook);
+				await this.fireBookRefresh(targetBook);
 				if (sourceBook) {
 					// refresh source book model to pick up latest changes
-					this.fireBookRefresh(sourceBook);
+					await this.fireBookRefresh(sourceBook);
```

Comment:
I just noticed something bad too - you have a debounce on this function but the debounce decorator doesn't take into account parameters. So if you call `fireBookRefresh` twice quickly with two separate books it'll only refresh one of them (the second one).

Couple ways to fix this : 

1. Remove debounce - not sure of perf/race condition issues if you do this
2. Move function into BookModel class and then have the debounc on that (since you won't need the parameter at that point). You could also probably get away with putting the debounce on the `initializeContents` call instead (although you should double check that doesn't mess up the .then

Reply:
I don't think that removing the debounce would be the best approach, because every time that the toc file is modified, it'll initilizeContents of the book. 
Moving the initialize method to the BookModel wouldn't let us use the vscode event for updating the tree view.
I think I'll call the initialize contents directly on the edit method and trigger the vscode event there as well.